### PR TITLE
BUG: masked mean unnecessarily overflowing

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Do we want to add mean to the Masked Array interface (I would say yes, but want to check anyway)? This would simplify things a bit.

cc @jorisvandenbossche

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Ah, nice, this is possible now because we require numpy >= 1.20 (the `where` keyword was only added later to `mean`)

> Do we want to add mean to the Masked Array interface (I would say yes, but want to check anyway)? This would simplify things a bit.

Yeah, I am not really sure why historically sum, prod, etc are methods, while mean is only handled inside `_reduce`

**Comment by phofl** [comment]
Yes, numpy now supports this in every version we need.

Ok great, confused me a little bit too

**Comment by phofl** [comment]
Added as method

File: pandas/tests/extension/base/dim2.py [hunk 1]
```diff
@@ -191,7 +192,11 @@ def test_reductions_2d_axis0(self, data, method):
             kwargs["ddof"] = 0
 
         try:
-            result = getattr(arr2d, method)(axis=0, **kwargs)
+            if method == "mean":
```
Comment (phofl) [inline pandas/tests/extension/base/dim2.py:195]: We have 2 all NA rows in our array that causes numpy to raise a RuntimeWarning with axis=0

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

**Pull request merged**